internal/js-dependency-scanner/src/main/java/org/wso2/security/tools/scanner/dependency/js/utils/ConfigParser.java

```diff
@@ -0,0 +1,248 @@
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ * /
+ */
+
+package org.wso2.security.tools.scanner.dependency.js.utils;
+
+
+import org.apache.log4j.Logger;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.wso2.security.tools.scanner.dependency.js.constants.IssueCreatorConstants;
+import org.wso2.security.tools.scanner.dependency.js.constants.JSScannerConstants;
+import org.wso2.security.tools.scanner.dependency.js.exception.ConfigParserException;
+import org.wso2.security.tools.scanner.dependency.js.issuecreator.IssueCreator;
+import org.wso2.security.tools.scanner.dependency.js.issuecreator.JIRAIssueCreator;
+import org.wso2.security.tools.scanner.dependency.js.issuecreator.JIRARestClient;
+import org.wso2.security.tools.scanner.dependency.js.model.Product;
+import org.wso2.security.tools.scanner.dependency.js.preprocessor.AtuwaDownloader;
+import org.wso2.security.tools.scanner.dependency.js.reportpublisher.GitUploader;
+import org.wso2.security.tools.scanner.dependency.js.reportpublisher.ReportUploader;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Properties;
+import java.util.Random;
+
+/**
+ * Util class to parse configuration properties. There are multiple configuration files related to each product and
+ * credentials related configuration files. Product related configuration files are parsed during preprocessing stage
+ * and credentials related config files are passed at the time where it is to be used.
+ */
+public class ConfigParser {
+
+    private static final Logger log = Logger.getLogger(ConfigParser.class);
+    private static Random random = new Random();
+
+    private ConfigParser() {
+    }
+
+    /**
+     * Parse the list of products which supports for weekly scanning.
+     *
+     * @return list of product names.
+     * @throws ConfigParserException exception occurred while parsing configuration details.
+     */
+    public static List<String> parseProductList() throws ConfigParserException {
+        List<String> products;
+        Properties properties = new Properties();
+        ClassLoader classLoader = ConfigParser.class.getClassLoader();
+        //load a properties file from class path
+        try (InputStream input = classLoader.getResourceAsStream(JSScannerConstants.PRODUCT_LIST_FILE)) {
+            properties.load(input);
+        } catch (IOException e) {
+            throw new ConfigParserException("Error occurred in parsing productList config file : " + e);
+        }
+        products = Arrays.asList(properties.getProperty(JSScannerConstants.PRODUCTS).
+                split(JSScannerConstants.PROPERTIES_FILE_DELIMETER));
```

Comment:
This should go inside the try block, because this depends on property loading being successful.